In [1]:
import pandas as pd
import numpy as np

## 행정동 데이터 전처리

In [21]:
import dbf

d = dbf.Table('../data/01_유동인구데이터(SK텔레콤)/4개지역_행정동.dbf')
d.open()
dbf.export(d, filename = '../data/01_유동인구데이터(SK텔레콤)/4개지역_행정동.csv',
           format = 'csv', header = True)

69

In [60]:
dongs = pd.read_csv('../data/01_유동인구데이터(SK텔레콤)/4개지역_행정동.csv', encoding = 'cp949')
dongs.head()

(b'STD_YYYY', 8)  (b'HDONG_CD', 8)   (b'HDONG_NM', 8)  (b'SIDO_CD', 7)  \
0              2020        2726067000  고산1동                            27   
1              2020        2726068000  고산2동                            27   
2              2020        2726069000  고산3동                            27   
3              2020        2726064000  두산동                             27   
4              2020        2726055000  만촌1동                            27   

   (b'SIDO_NM', 7)  (b'SGNG_CD', 7)    (b'SGNG_NM', 7)  (b'AREA', 4)  \
0  대구광역시                      27260  수성구                  8881312.38   
1  대구광역시                      27260  수성구                 24523329.28   
2  대구광역시                      27260  수성구                  5315712.45   
3  대구광역시                      27260  수성구                  1966650.15   
4  대구광역시                      27260  수성구                  2830625.94   

   (b'LENGTH', 6)  (b'X_COORD', 7)  (b'Y_COORD', 7)  
0        17757.54       128.699631        35.811747  
1        25763.81       128.677147        35.838758  
2        11274.68       128.714044        35.849048  
3         7777.10       128.620833        35.828566  
4         8778.85       128.649513        35.870682

In [61]:
col = ['STD_YYYY', 'HDONG_CF', 'HDONG_NM', 'SIDO_CD', 'SIDO_NM', 'SGNG_CD', 'SGNG_NM', 'AREA', 'LENGTH', 'LNG', 'LAT']
dongs.columns = col
dongs.head()

STD_YYYY    HDONG_CF           HDONG_NM  SIDO_CD          SIDO_NM  SGNG_CD  \
0      2020  2726067000  고산1동                    27  대구광역시              27260   
1      2020  2726068000  고산2동                    27  대구광역시              27260   
2      2020  2726069000  고산3동                    27  대구광역시              27260   
3      2020  2726064000  두산동                     27  대구광역시              27260   
4      2020  2726055000  만촌1동                    27  대구광역시              27260   

             SGNG_NM         AREA    LENGTH         LNG        LAT  
0  수성구                 8881312.38  17757.54  128.699631  35.811747  
1  수성구                24523329.28  25763.81  128.677147  35.838758  
2  수성구                 5315712.45  11274.68  128.714044  35.849048  
3  수성구                 1966650.15   7777.10  128.620833  35.828566  
4  수성구                 2830625.94   8778.85  128.649513  35.870682

In [62]:
dongs['HDONG_NM'].unique()[:5]

array(['고산1동             ', '고산2동             ', '고산3동             ',
       '두산동              ', '만촌1동             '], dtype=object)

**데이터 전처리의 편의를 위해 object로 되어있는 value값에서 space값을 먼저 처리해주자.** 

In [63]:
dong_name = []
si_name = []
gu_name = []

for i in range(len(dongs['HDONG_NM'])):
    dong_name.append(dongs['HDONG_NM'][i].replace(' ', ''))
    si_name.append(dongs['SIDO_NM'][i].replace(' ', ''))
    gu_name.append(dongs['SGNG_NM'][i].replace(' ', ''))

In [64]:
dongs['HDONG_NM'] = dong_name
dongs['SIDO_NM'] = si_name
dongs['SGNG_NM'] = gu_name

In [65]:
dongs.head()

STD_YYYY    HDONG_CF HDONG_NM  SIDO_CD SIDO_NM  SGNG_CD SGNG_NM  \
0      2020  2726067000     고산1동       27   대구광역시    27260     수성구   
1      2020  2726068000     고산2동       27   대구광역시    27260     수성구   
2      2020  2726069000     고산3동       27   대구광역시    27260     수성구   
3      2020  2726064000      두산동       27   대구광역시    27260     수성구   
4      2020  2726055000     만촌1동       27   대구광역시    27260     수성구   

          AREA    LENGTH         LNG        LAT  
0   8881312.38  17757.54  128.699631  35.811747  
1  24523329.28  25763.81  128.677147  35.838758  
2   5315712.45  11274.68  128.714044  35.849048  
3   1966650.15   7777.10  128.620833  35.828566  
4   2830625.94   8778.85  128.649513  35.870682

In [66]:
dongs['HDONG_NM'].unique()[:5]

array(['고산1동', '고산2동', '고산3동', '두산동', '만촌1동'], dtype=object)

**최근 기준으로 서울특별시, 대구광역시 각각 노원구, 중구, 수성구, 중구의 면적, 길이, 위도, 경도 데이터이므로 연도는 필요없다고 판단. 즉, 'STD_YYYY'데이터를 드랍한다.**

In [67]:
dongs.drop(['STD_YYYY'], axis = 1, inplace = True)

### 코드와 이름 매칭

In [68]:
dongs['SIDO_NM'].unique()

array(['대구광역시', '서울특별시'], dtype=object)

* 'SIDO_NM'이 27이면 대구광역시, 11이면 서울특별시를 나타내고 있다. 

In [69]:
dongs['SGNG_NM'].unique()

array(['수성구', '중구', '노원구'], dtype=object)

In [70]:
dongs[dongs['SGNG_NM'] == '수성구'].iloc[0:1, :]

HDONG_CF HDONG_NM  SIDO_CD SIDO_NM  SGNG_CD SGNG_NM        AREA  \
0  2726067000     고산1동       27   대구광역시    27260     수성구  8881312.38   

     LENGTH         LNG        LAT  
0  17757.54  128.699631  35.811747

In [71]:
dongs[dongs['SGNG_NM'] == '노원구'].iloc[0:1, :]

HDONG_CF HDONG_NM  SIDO_CD SIDO_NM  SGNG_CD SGNG_NM        AREA  \
35  1135059500     공릉1동       11   서울특별시    11350     노원구  1224521.56   

     LENGTH         LNG        LAT  
35  5518.75  127.072581  37.624093

In [72]:
dongs[dongs['SGNG_NM'] == '중구']

HDONG_CF HDONG_NM  SIDO_CD SIDO_NM  SGNG_CD SGNG_NM        AREA  \
23  2711064000     남산1동       27   대구광역시    27110      중구   379731.51   
24  2711065000     남산2동       27   대구광역시    27110      중구   371772.72   
25  2711066000     남산3동       27   대구광역시    27110      중구   408260.33   
26  2711067000     남산4동       27   대구광역시    27110      중구   449545.07   
27  2711068000     대봉1동       27   대구광역시    27110      중구   565841.23   
28  2711069000     대봉2동       27   대구광역시    27110      중구   273813.87   
29  2711059500      대신동       27   대구광역시    27110      중구   500437.75   
30  2711051700      동인동       27   대구광역시    27110      중구   985061.69   
31  2711054500      삼덕동       27   대구광역시    27110      중구   643036.30   
32  2711056500     성내1동       27   대구광역시    27110      중구   757321.16   
33  2711057500     성내2동       27   대구광역시    27110      중구   985981.30   
34  2711058500     성내3동       27   대구광역시    27110      중구   754903.81   
54  1114059000      광희동       11   서울특별시    11140      중구   741195.05   
55  1114062500      다산동       11   서울특별시    11140      중구   521341.19   
56  1114066500      동화동       11   서울특별시    11140      중구   328213.50   
57  1114055000       명동       11   서울특별시    11140      중구  1053644.29   
58  1114052000      소공동       11   서울특별시    11140      중구   893419.62   
59  1114065000     신당5동       11   서울특별시    11140      중구   321347.03   
60  1114061500      신당동       11   서울특별시    11140      중구   544408.19   
61  1114063500      약수동       11   서울특별시    11140      중구   490077.51   
62  1114060500     을지로동       11   서울특별시    11140      중구   599091.52   
63  1114058000      장충동       11   서울특별시    11140      중구  1310554.91   
64  1114068000      중림동       11   서울특별시    11140      중구   482329.92   
65  1114064500      청구동       11   서울특별시    11140      중구   336949.37   
66  1114057000       필동       11   서울특별시    11140      중구  1173378.79   
67  1114067000      황학동       11   서울특별시    11140      중구   328678.23   
68  1114054000      회현동       11   서울특별시    11140      중구   847351.57   

     LENGTH         LNG        LAT  
23  3548.70  128.592666  35.859950  
24  2888.45  128.588268  35.864118  
25  2582.10  128.585984  35.859877  
26  2892.57  128.579200  35.860834  
27  3410.40  128.604535  35.859377  
28  3255.92  128.598556  35.857710  
29  3493.32  128.578949  35.867271  
30  4251.23  128.607589  35.870268  
31  4041.38  128.604447  35.865187  
32  5292.34  128.597171  35.868420  
33  4871.69  128.589308  35.870756  
34  4586.72  128.582450  35.874950  
54  5359.72  127.003693  37.565177  
55  4124.64  127.008419  37.555034  
56  2994.67  127.018648  37.560674  
57  6472.97  126.984205  37.563586  
58  4367.79  126.974020  37.564293  
59  2668.71  127.021838  37.563627  
60  3475.62  127.013671  37.565724  
61  3896.34  127.010301  37.550170  
62  4524.58  126.996387  37.566585  
63  7373.04  127.001637  37.555864  
64  3500.86  126.966042  37.557356  
65  2696.70  127.015201  37.556863  
66  6575.62  126.992277  37.556817  
67  2381.97  127.020846  37.568536  
68  4540.89  126.976856  37.557178

* 'SGNG_NM'이 27260이면 대구광역시의 수성구를, 11350이면 서울특별시의 노원구를, 27110이면 대구광역시의 중구를, 11140이면 서울특별시의 중구를 나타내고 있음을 알 수 있다.

### 행정동 데이터 필요데이터로 만들어주기

**시도코드를 알아냈으므로 위의 데이터에서 필요한 Area, Length, Longitude, Latitude만을 따와서 하나의 데이터 파일로 만들자.**

In [73]:
region = []

for i in range(len(dongs['HDONG_NM'])) :
    region.append(dongs['SIDO_NM'][i] + ' ' + dongs['SGNG_NM'][i] + ' ' + dongs['HDONG_NM'][i])

In [74]:
dongs = dongs.iloc[:, 6:]

In [75]:
dongs['지역명'] = region 
dongs.index = dongs['지역명']
dongs.drop(['지역명'], axis = 1, inplace = True)
dongs.reset_index(drop = False, inplace = True)

In [76]:
col = ['지역명', '면적', '길이', '위도', '경도']
dongs.columns = col
dongs.head()

지역명           면적        길이          위도         경도
0  대구광역시 수성구 고산1동   8881312.38  17757.54  128.699631  35.811747
1  대구광역시 수성구 고산2동  24523329.28  25763.81  128.677147  35.838758
2  대구광역시 수성구 고산3동   5315712.45  11274.68  128.714044  35.849048
3   대구광역시 수성구 두산동   1966650.15   7777.10  128.620833  35.828566
4  대구광역시 수성구 만촌1동   2830625.94   8778.85  128.649513  35.870682

In [77]:
dongs.head()

지역명           면적        길이          위도         경도
0  대구광역시 수성구 고산1동   8881312.38  17757.54  128.699631  35.811747
1  대구광역시 수성구 고산2동  24523329.28  25763.81  128.677147  35.838758
2  대구광역시 수성구 고산3동   5315712.45  11274.68  128.714044  35.849048
3   대구광역시 수성구 두산동   1966650.15   7777.10  128.620833  35.828566
4  대구광역시 수성구 만촌1동   2830625.94   8778.85  128.649513  35.870682

In [164]:
dongs.to_excel('../data/행정동(면적_길이_위도_경도).xlsx', index = False)

### 시도 코드
* 'SIDO_NM' -> 27 : 대구광역시
* 'SIDO_NM' -> 11 : 서울특별시
* 'SGNG_NM' -> 27260 : 대구광역시 수성구
* 'SGNG_NM' -> 27110 : 대구광역시 중구
* 'SGNG_NM' -> 11350 : 서울특별시 노원구
* 'SGNG_NM' -> 11140 : 서울특별시 중구

##  Flow 데이터 - 전처리 함수 만들기!

In [152]:
flow1 = pd.read_csv('../data/01_유동인구데이터(SK텔레콤)/4개지역_FLOW_AGE_201902.csv', sep = '|')
flow1

STD_YM   STD_YMD    HDONG_CD HDONG_NM  MAN_FLOW_POP_CNT_0004  \
0     201902  20190201  1114052000      소공동                  16.64   
1     201902  20190202  1114052000      소공동                   3.14   
2     201902  20190203  1114052000      소공동                   1.38   
3     201902  20190204  1114052000      소공동                   0.08   
4     201902  20190205  1114052000      소공동                   0.00   
...      ...       ...         ...      ...                    ...   
1927  201902  20190224  2726069000     고산3동                   0.05   
1928  201902  20190225  2726069000     고산3동                   0.05   
1929  201902  20190226  2726069000     고산3동                   0.03   
1930  201902  20190227  2726069000     고산3동                   0.08   
1931  201902  20190228  2726069000     고산3동                   0.19   

      MAN_FLOW_POP_CNT_0509  MAN_FLOW_POP_CNT_1014  MAN_FLOW_POP_CNT_1519  \
0                    116.18                 419.70                1553.57   
1                    141.07                 763.30                1405.29   
2                    120.39                 698.23                1117.17   
3                    141.61                 772.41                1248.85   
4                    254.70                1394.92                2190.69   
...                     ...                    ...                    ...   
1927                 146.86                 677.67                1084.16   
1928                 187.83                 534.66                1024.18   
1929                 146.97                 494.97                 925.04   
1930                 164.30                 555.82                1027.75   
1931                 173.51                 508.44                1107.87   

      MAN_FLOW_POP_CNT_2024  MAN_FLOW_POP_CNT_2529  ...  \
0                   5196.07               11195.62  ...   
1                   4235.82                7592.34  ...   
2                   3032.90                5173.35  ...   
3                   3001.95                4641.82  ...   
4                   3420.15                5478.30  ...   
...                     ...                    ...  ...   
1927                1545.11                1952.14  ...   
1928                1490.25                1840.75  ...   
1929                1413.57                1748.78  ...   
1930                1437.50                1798.46  ...   
1931                1473.76                1977.00  ...   

      WMAN_FLOW_POP_CNT_2529  WMAN_FLOW_POP_CNT_3034  WMAN_FLOW_POP_CNT_3539  \
0                   13323.19                12741.15                11628.80   
1                    7898.27                 6580.27                 5714.75   
2                    5079.63                 4454.43                 3905.32   
3                    4317.63                 3676.47                 3506.37   
4                    4813.86                 4967.92                 5161.96   
...                      ...                     ...                     ...   
1927                 1502.39                 1492.40                 1980.33   
1928                 1419.83                 1502.75                 2289.69   
1929                 1472.12                 1493.60                 2205.07   
1930                 1436.20                 1433.72                 2195.25   
1931                 1559.67                 1510.90                 2269.63   

      WMAN_FLOW_POP_CNT_4044  WMAN_FLOW_POP_CNT_4549  WMAN_FLOW_POP_CNT_5054  \
0                   10006.80                 9807.06                 8099.38   
1                    4946.22                 5055.32                 4560.27   
2                    3484.47                 3703.24                 3270.26   
3                    3108.99                 3324.07                 2841.35   
4                    4856.63                 5212.69                 4462.94   
...                      ...                     ...                     ...   
1927      

* STD_YM과 STD_YMD는 D가 추가되어 있는 형태이므로 STD_YM을 드랍한다.
* HDONG_CD으로 부터 광역시 및 구를 알아낸 후 HDONG_NM과 합쳐 새로운 Column을 만들어준다.
* STD_YMD를 '기준년월일'로 새로운 column을 '지역명'으로 만들고 유동인구 앞쪽의 columns은 전부 드랍해준다.
* 유동인구 데이터는 '00세~04세'와 같은 형식으로 데이터를 변경해주도록 하자. 

#### STD_YM드랍 및 이름변경

In [153]:
flow1.drop(['STD_YM'], axis = 1, inplace = True)
flow1.rename(columns = {'STD_YMD' : '기준년월일'}, inplace = True)

#### 지역명 column만들기

In [154]:
print(str(flow1['HDONG_CD'][0])[:2])
print(str(flow1['HDONG_CD'][0])[:5])

11
11140


In [155]:
si_name = []
gu_name = []

for i in range(len(flow1['HDONG_CD'])) :
    if str(flow1['HDONG_CD'][i])[:2] == '11' :
        si_name.append('서울광역시')
    else :
        si_name.append('대구광역시')
        
for i in range(len(flow1['HDONG_CD'])) :
    if str(flow1['HDONG_CD'][i])[:5] == '27260' :
        gu_name.append('수성구')
    elif str(flow1['HDONG_CD'][i])[:5] == '27110':
        gu_name.append('중구')
    elif str(flow1['HDONG_CD'][i])[:5] == '11350' :
        gu_name.append('노원구')
    else :
        gu_name.append('중구')

In [156]:
region = []

for i in range(len(flow1['HDONG_CD'])) :
    region.append(si_name[i] + ' ' + gu_name[i] + ' ' + flow1['HDONG_NM'][i])

In [157]:
flow1['지역명'] = region
flow1.index = flow1['지역명']
flow1.drop(['지역명'], axis = 1, inplace = True)
flow1.reset_index(drop =  False, inplace = True)

In [158]:
flow1.drop(['HDONG_CD', 'HDONG_NM'], axis = 1, inplace = True)

In [159]:
flow1.head()

지역명     기준년월일  MAN_FLOW_POP_CNT_0004  MAN_FLOW_POP_CNT_0509  \
0  서울광역시 중구 소공동  20190201                  16.64                 116.18   
1  서울광역시 중구 소공동  20190202                   3.14                 141.07   
2  서울광역시 중구 소공동  20190203                   1.38                 120.39   
3  서울광역시 중구 소공동  20190204                   0.08                 141.61   
4  서울광역시 중구 소공동  20190205                   0.00                 254.70   

   MAN_FLOW_POP_CNT_1014  MAN_FLOW_POP_CNT_1519  MAN_FLOW_POP_CNT_2024  \
0                 419.70                1553.57                5196.07   
1                 763.30                1405.29                4235.82   
2                 698.23                1117.17                3032.90   
3                 772.41                1248.85                3001.95   
4                1394.92                2190.69                3420.15   

   MAN_FLOW_POP_CNT_2529  MAN_FLOW_POP_CNT_3034  MAN_FLOW_POP_CNT_3539  ...  \
0               11195.62               14249.35               16995.99  ...   
1                7592.34                7259.61                6721.80  ...   
2                5173.35                5067.98                4761.52  ...   
3                4641.82                4508.23                4477.35  ...   
4                5478.30                5735.30                6318.20  ...   

   WMAN_FLOW_POP_CNT_2529  WMAN_FLOW_POP_CNT_3034  WMAN_FLOW_POP_CNT_3539  \
0                13323.19                12741.15                11628.80   
1                 7898.27                 6580.27                 5714.75   
2                 5079.63                 4454.43                 3905.32   
3                 4317.63                 3676.47                 3506.37   
4                 4813.86                 4967.92                 5161.96   

   WMAN_FLOW_POP_CNT_4044  WMAN_FLOW_POP_CNT_4549  WMAN_FLOW_POP_CNT_5054  \
0                10006.80                 9807.06                 8099.38   
1                 4946.22                 5055.32                 4560.27   
2                 3484.47                 3703.24                 3270.26   
3                 3108.99                 3324.07                 2841.35   
4                 4856.63                 5212.69                 4462.94   

   WMAN_FLOW_POP_CNT_5559  WMAN_FLOW_POP_CNT_6064  WMAN_FLOW_POP_CNT_6569  \
0                 7260.17                 5757.53                 3359.85   
1                 4364.57                 3838.77                 2453.97   
2                 2984.05                 2484.98                 1511.67   
3                 2644.42                 2164.35                 1230.21   
4                 4183.38                 3265.62                 1692.74   

   WMAN_FLOW_POP_CNT_70U  
0                4431.71  
1                3207.26  
2                1954.98  
3                1520.35  
4                2147.09  

[5 rows x 32 columns]

In [160]:
age = ['지역명', '기준년월일']

for i in range(len(flow1.columns)) :
    if flow1.columns[i][-1] == 'U' and flow1.columns[i][0] == 'M':
        age.append('남자 70세 이상')
        
    elif flow1.columns[i][0] == 'M' :
        age.append('남자' + ' ' + str(flow1.columns[i][-4:-2]) + '세~' + str(flow1.columns[i][-2:]) + '세')
        
    elif flow1.columns[i][-1] == 'U' and flow1.columns[i][0] == 'W' :
        age.append('여자 70세 이상')
        
    elif flow1.columns[i][0] == 'W' :
        age.append('여자' + ' ' + str(flow1.columns[i][-4:-2]) + '세~' + str(flow1.columns[i][-2:]) + '세')

In [161]:
flow1.columns = age

In [162]:
flow1.head()

지역명     기준년월일  남자 00세~04세  남자 05세~09세  남자 10세~14세  남자 15세~19세  \
0  서울광역시 중구 소공동  20190201       16.64      116.18      419.70     1553.57   
1  서울광역시 중구 소공동  20190202        3.14      141.07      763.30     1405.29   
2  서울광역시 중구 소공동  20190203        1.38      120.39      698.23     1117.17   
3  서울광역시 중구 소공동  20190204        0.08      141.61      772.41     1248.85   
4  서울광역시 중구 소공동  20190205        0.00      254.70     1394.92     2190.69   

   남자 20세~24세  남자 25세~29세  남자 30세~34세  남자 35세~39세  ...  여자 25세~29세  \
0     5196.07    11195.62    14249.35    16995.99  ...    13323.19   
1     4235.82     7592.34     7259.61     6721.80  ...     7898.27   
2     3032.90     5173.35     5067.98     4761.52  ...     5079.63   
3     3001.95     4641.82     4508.23     4477.35  ...     4317.63   
4     3420.15     5478.30     5735.30     6318.20  ...     4813.86   

   여자 30세~34세  여자 35세~39세  여자 40세~44세  여자 45세~49세  여자 50세~54세  여자 55세~59세  \
0    12741.15    11628.80    10006.80     9807.06     8099.38     7260.17   
1     6580.27     5714.75     4946.22     5055.32     4560.27     4364.57   
2     4454.43     3905.32     3484.47     3703.24     3270.26     2984.05   
3     3676.47     3506.37     3108.99     3324.07     2841.35     2644.42   
4     4967.92     5161.96     4856.63     5212.69     4462.94     4183.38   

   여자 60세~64세  여자 65세~69세  여자 70세 이상  
0     5757.53     3359.85    4431.71  
1     3838.77     2453.97    3207.26  
2     2484.98     1511.67    1954.98  
3     2164.35     1230.21    1520.35  
4     3265.62     1692.74    2147.09  

[5 rows x 32 columns]

### 함수

In [177]:
def open_file(file_name) :
    name = pd.read_csv('../data/01_유동인구데이터(SK텔레콤)/%s' % file_name, sep = '|')
    return name

In [178]:
def STD_YM_drop(file) :
    file.drop(['STD_YM'], axis = 1, inplace = True)
    file.rename(columns = {'STD_YMD' : '기준년월일'}, inplace = True)
    return file

In [196]:
def make_new_column(file) :
    si_name = []
    gu_name = []
    
    for i in range(len(file)) :
        if str(file['HDONG_CD'][i])[:2] == '11' :
            si_name.append('서울광역시')
        else :
            si_name.append('대구광역시')
            
    for i in range(len(file['HDONG_CD'])) :
        if str(file['HDONG_CD'][i])[:5] == '27260' :
            gu_name.append('수성구')
        elif str(file['HDONG_CD'][i])[:5] == '27110':
            gu_name.append('중구')
        elif str(file['HDONG_CD'][i])[:5] == '11350' :
            gu_name.append('노원구')
        else :
            gu_name.append('중구')
            
    region = []
    
    for i in range(len(file)) :
        region.append(si_name[i] + ' ' + gu_name[i] + ' '+ file['HDONG_NM'][i])
        
    file['지역명'] = region
    file.index = file['지역명']
    file.drop(['지역명'], axis = 1, inplace = True)
    file.reset_index(drop = False, inplace  = True)
    
    return file

In [180]:
def drop_columns(file) :
    file.drop(['HDONG_CD', 'HDONG_NM'], axis = 1, inplace = True)
    return file

In [181]:
def make_columns_name(file) :
    age = ['지역명', '기준년월일']

    for i in range(len(file.columns)) :
        if file.columns[i][-1] == 'U' and file.columns[i][0] == 'M':
            age.append('남자 70세 이상')
        
        elif file.columns[i][0] == 'M' :
            age.append('남자' + ' ' + str(file.columns[i][-4:-2]) + '세~' + str(file.columns[i][-2:]) + '세')
        
        elif file.columns[i][-1] == 'U' and file.columns[i][0] == 'W' :
            age.append('여자 70세 이상')
        
        elif file.columns[i][0] == 'W' :
            age.append('여자' + ' ' + str(file.columns[i][-4:-2]) + '세~' + str(file.columns[i][-2:]) + '세')
            
    file.columns = age
    
    return file

In [165]:
def save_file(file, name) :
    file.to_excel('../data/%s' % name, index = False)

### 파일 만들기

In [204]:
import glob

In [205]:
path = '../data/01_유동인구데이터(SK텔레콤)'

file_list = glob.glob(path + '\\*.csv')
file_list

['../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_201902.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_201903.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_201904.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_201905.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_202002.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_202003.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_202004.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_AGE_202005.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_201902.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_201903.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_201904.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_201905.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_202002.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_202003.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_202004.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_FLOW_TIME_202005.CSV',
 '../data/01_유동인구데이터(SK텔레콤)\\4개지역_행정동.csv']

In [264]:
print(file_list[0][26:46])
print(len(file_list[0][26:46]))

4개지역_FLOW_AGE_201902
20


In [219]:
naming = []

for i in range(len(file_list)) :
    if len(file_list[i][26:46]) == 20:
        naming.append(file_list[i][26:46])

In [265]:
naming = naming[:8]
naming

['4개지역_FLOW_AGE_201902',
 '4개지역_FLOW_AGE_201903',
 '4개지역_FLOW_AGE_201904',
 '4개지역_FLOW_AGE_201905',
 '4개지역_FLOW_AGE_202002',
 '4개지역_FLOW_AGE_202003',
 '4개지역_FLOW_AGE_202004',
 '4개지역_FLOW_AGE_202005']

In [266]:
print(naming[0][-6:-2])
print(naming[0][-2:])

2019
02


In [262]:
def file_parsing(file_list) :
    
    name_list = pd.DataFrame(file_list)
    name_list = name_list + '.csv'
    name_list = name_list.iloc[:, 0]
    
    for i in range(len(name_list)) :
        flow = open_file(str(name_list[i]))
        flow = STD_YM_drop(flow)
        flow = make_new_column(flow)
        flow = drop_columns(flow)
        flow = make_columns_name(flow)
        
        save_file(flow, '유동인구_나이{}년{}월.xlsx'.format(file_list[i][-6:-2], file_list[i][-2:]))

In [263]:
file_parsing(naming)